src/http-client/index.ts

```diff
@@ -38,13 +40,15 @@ export interface HTTPClient {
   request(req: HTTPRequest): Promise<HTTPResponse>;
 }
 
-export const getDefaultHTTPClient = () => {
-  // WIP: we only have one implementation right now, but should eventually
-  // inspect the environment for the correct implementation
-  return new FetchClient();
-};
+export const getDefaultHTTPClient = () =>
+  isNode() ? NodeHTTP2Client.getClient() : new FetchClient();
 
 // utility functions
 
 export const isHTTPResponse = (res: any): res is HTTPResponse =>
   res instanceof Object && "body" in res && "headers" in res && "status" in res;
+
+const isNode = () =>
+  Object.prototype.toString.call(
+    typeof process !== "undefined" ? process : 0
+  ) === "[object process]";
```

Comment:
I haven't looked at this whole PR yet - so more may come later! But I think we can be a bit tighter here:

```suggestion
const isNode = () =>  (process !== undefined && process.release === "node");
```

Example:

```
 > process.release
{ name: 'node', lt
  name: 'node',
  lts: 'Hydrogen',
  sourceUrl: 'https://nodejs.org/download/release/v18.12.1/node-v18.12.1.tar.gz',
  headersUrl: 'https://nodejs.org/download/release/v18.12.1/node-v18.12.1-headers.tar.gz'
}
 > 
```